snapshots/lcow/lcow.go

```diff
@@ -378,8 +412,38 @@ func (s *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 	return s.mounts(newSnapshot), nil
 }
 
-func (s *snapshotter) openOrCreateScratch(ctx context.Context, sizeGB int) (_ *os.File, err error) {
-	// Create the scratch.vhdx cache file if it doesn't already exist.
+func (s *snapshotter) handleSharing(ctx context.Context, id, snDir string) error {
+	var key string
+	if err := s.Walk(ctx, func(ctx context.Context, info snapshots.Info) error {
+		if strings.Contains(info.Name, id) {
+			key = info.Name
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	mounts, err := s.Mounts(ctx, key)
+	if err != nil {
+		return errors.Wrap(err, "failed to get mounts for owner snapshot")
+	}
+
+	sandboxPath := filepath.Join(mounts[0].Source, "sandbox.vhdx")
+	linkPath := filepath.Join(snDir, "sandbox.vhdx")
+	if _, err := os.Stat(sandboxPath); err != nil {
+		return errors.Wrap(err, "failed to find sandbox.vhdx in snapshot directory")
+	}
+
+	// We've found everything we need, now just make a symlink in our new snapshot to the
+	// sandbox.vhdx in the scratch we're asking to share.
+	if err := os.Symlink(sandboxPath, linkPath); err != nil {
```

Comment:
Why? What is the purpose of the symlink?

Reply:
So we can find the disk we're sharing in the shim. When we're reading the layer folders in the shim, we'd eval the symlink and see that it points to a disk we've already mounted, and then we can just use the refcounting we have on scsi mounts to handle removal/not actually remounting it haha.

Reply:
Isn't it just simpler to assume if there is no sandbox.vhdx in the writable snapshot folder and its not using readonly rootfs then clearly we are trying to re-use the sandbox of the pod? I dont see why we really need the level of complication here but its your code :)

Reply:
The reason is because we need to know where the pods scratch was mounted for future containers in the shim. In [MountContainerLayers](https://github.com/microsoft/hcsshim/blob/1286fcd6c97490c14ce152e8e7cac62ef5d1b38c/internal/layers/layers.go#L64), we'd use where the disk was mounted in the UVM to decide where to setup the overlayfs for any containers that want to re-use this space. During this call there's no concept of a pod or if the container we're mounting the layers for belongs to one, and I don't think we should plumb a bunch of stuff through to make this call aware of that. This seemed like the simplest way to be able to tell where the pods scratch was mounted so we can use the mountpoint to setup more than one overlay using the same space. 

e.g. Follow the symlink -> Try to add scsi mount -> See that it was already added and get back an object that has where it was already mounted -> Use this to setup new overlays in the same mount point.